### PR TITLE
Develop Code Review

### PR DESCRIPTION
**Pull request by rennergade**
I created this branch based on the last commit before we started to compare our changes to develop, let's code review it here.

### CODE REVIEW

File: seattlelib/lind_fs_constants.py [hunk 1]
```diff
@@ -113,6 +113,11 @@
 MAX_FD = 1024
 STARTINGFD = 0
 
+#largest pipe
+STARTING_PIPE = 0
+MAX_PIPE = MAX_FD
```
Comment (rennergade) [inline seattlelib/lind_fs_constants.py]: I think we discussed that having the maximum amount of pipes equal to the max FD number is actually a bad idea, but I'm not sure what assumptions we should make. @JustinCappos ?

File: seattlelib/lind_fs_calls.py [hunk 2]
```diff
@@ -157,16 +166,37 @@ class SyscallError(Exception):
 class UnimplementedError(Exception):
   """A call was called with arguments that are not fully implemented"""
 
+def repy_deepcopy(obj):
```
Comment (rennergade) [inline seattlelib/lind_fs_calls.py]: Would be nice to have a 1-2 line summary comment, but in general this code is great.

File: seattlelib/lind_fs_calls.py [hunk 3]
```diff
@@ -396,1725 +413,2042 @@ def _get_absolute_path(path):
...
+      # If 0, remove the entry from the inode table if there are no open handles
+      # If open handles exist, flag it to unlink when the last handle is closed
+      if filesystemmetadata['inodetable'][thisinode]['linkcount'] == 0:
+        fdsforinode = _lookup_fds_by_inode(thisinode)
+        if len(fdsforinode) == 0:
+            del filesystemmetadata['inodetable'][thisinode]
+        else:
+            filesystemmetadata['inodetable'][thisinode]['unlinked'] = True
 
+        # TODO: I also would remove the file.   However, I need to do special
```
Comment (rennergade) [inline seattlelib/lind_fs_calls.py]: Was this covered in the above changes?

Comment (jesings) [reply seattlelib/lind_fs_calls.py]: Yeah this should have been covered. Good catch. Let's remove before merge.

**Comment by rennergade** [comment]
This PR was only used for reference and is now closed. I'll now be making develop the new master and archiving the old master branch.

**Pull request closed without merging**